hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueStore.java

```diff
@@ -34,14 +36,34 @@
  */
 public interface UniqueStore extends TokenStore {
 
-	ResponseCodeEnum mint(final TokenID tId, String memo, RichInstant creationTime);
 
-	//	MerkleUniqueToken getUnique(final EntityId eId, final int serialNum);
-//	Iterator<MerkleUniqueTokenId> getByToken(final MerkleUniqueToken token);
-//	Iterator<MerkleUniqueTokenId> getByTokenFromIdx(final MerkleUniqueToken token, final int start);
-//	Iterator<MerkleUniqueTokenId> getByTokenFromIdxToIdx(final MerkleUniqueToken token, final int start, final int end);
-//	Iterator<MerkleUniqueTokenId> getByAccountFromIdxToIdx(final AccountID aId, final int start, final int end);
+	/**
+	 * Attempts to mint unique tokens in a batch. Rollbacks the batch if anything goes wrong
+	 *
+	 * @param txBody {@link TokenMintTransactionBody} the body of the mint tx
+	 * @param creationTime {@link RichInstant} the consensus time of the minting
+	 * @return {@link ResponseCodeEnum} OK on success, else - the cause of the fail
+	 */
+	ResponseCodeEnum mint(final TokenMintTransactionBody txBody, final RichInstant creationTime);
```

Comment:
This can be refactored to use `CreationResult` generic instead of `ResponseCodeEnum`. We can please the list of created NFTs in the `created` optional. That would remove the need for `List<Long> popLastMintedSerialNumbers();` and everything will be concentrated in this one `mint` method